pyaedt/maxwell.py

```diff
@@ -1460,15 +1460,104 @@ def assign_insulating(self, geometry_selection, insulation=None):
             "ElectroDCConduction",
         ]:
 
-            if not insulation:
-                insulation = generate_unique_name("Insulation")
-            elif insulation in self.modeler.get_boundaries_name():
-                insulation = generate_unique_name(insulation)
+            if not insulation_name:
+                insulation_name = generate_unique_name("Insulation")
+            elif insulation_name in self.modeler.get_boundaries_name():
+                insulation_name = generate_unique_name(insulation_name)
 
             listobj = self.modeler.convert_to_selections(geometry_selection, True)
             props = {"Objects": listobj}
 
-            return self._create_boundary(insulation, props, "Insulating")
+            return self._create_boundary(insulation_name, props, "Insulating")
+        return False
+
+    @pyaedt_function_handler()
+    def assign_impedance(
+        self,
+        geometry_selection,
+        use_material=False,
```

Comment:
@MaxJPRey what if we remove this boolean and simply check if material_name is passed and automatically set internally to True or False depending if the material is passed or not?

Reply:
That is an excellent idea. I will implement that.